File: code/backend/java/core/src/main/java/games/paths/core/model/story/TextInfo.java
package games.paths.core.model.story;

/**
 * TextInfo - Domain model for a resolved text entry.
 * Includes both short and long text, resolved language, and optional
 * copyright and creator information. Immutable record.
 *
 * <p>Added in Step 16 for the content detail APIs.</p>
 */
public record TextInfo(
        int idText,
        String lang,
        String resolvedLang,
        String shortText,
        String longText,
        String copyrightText,
        String linkCopyright,
        CreatorInfo creator) {
}

File: code/backend/php/src/Core/Domain/Story/TextInfo.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class TextInfo implements \JsonSerializable
{
    public function __construct(
        public int $idText,
        public string $lang,
        public string $resolvedLang,
        public ?string $shortText = null,
        public ?string $longText = null,
        public ?string $copyrightText = null,
        public ?string $linkCopyright = null,
        public ?CreatorInfo $creator = null
    ) {
    }

    public function jsonSerialize(): array
    {
        return [
            'idText' => $this->idText,
            'lang' => $this->lang,
            'resolvedLang' => $this->resolvedLang,
            'shortText' => $this->shortText,
            'longText' => $this->longText,
            'copyrightText' => $this->copyrightText,
            'linkCopyright' => $this->linkCopyright,
            'creator' => $this->creator,
        ];
    }
}
